app/helpers/i18n_helper.rb

```diff
@@ -49,20 +49,17 @@ def date_eras_gregorian(date)
 
   # Return the most accurate date from an array of dates using scoring algorithm below
   def date_most_accurate(dates)
-    return nil unless dates.present? && (dates.is_a?(Array) || dates.is_a?(String))
-    dates = [dates] if dates.is_a?(String)
-    most_accurate = nil
-    max_score = 0
-    dates.map { |date| date_eras_gregorian(date) || date.strip }.each_with_index do |date, index|
-      if index.zero?
-        max_score = date_score(date)
+    return nil unless dates.present?
+    return dates if dates.is_a?(String)
```

Comment:
This is a helper which in principle could be called from anywhere, not just from entity_date. Also, there are tests for passing nil and string, so these break as well if I remove the lines. Prefer to keep these sanity checks in place to ensure method contract is not violated.